pandas/sparse/tests/test_sparse.py

```diff
@@ -948,6 +948,23 @@ def setUp(self):
 
         self.empty = SparseDataFrame()
 
+    def test_fill_value_missing_with_binary_operators(self):
+        # GH12723
+        dat1 = np.arange(10, dtype='float')
+        df1 = DataFrame({'foo': dat1}, index=range(10)).to_sparse()
+
+        dat2 = dat1.copy()
+        dat2[5] = np.nan
+        df2 = DataFrame({'foo': dat2}, index=range(10)).to_sparse()
+
+        exp = df2.fillna(0).add(2)
+        res = df2.add(2, fill_value=0)
+        tm.assert_sp_frame_equal(res, exp)
+
+        exp = df2.fillna(0).add(df1)
+        res = df2.add(df1, fill_value=0)
+        tm.assert_sp_frame_equal(res, exp)
```

Comment:
I'm not familiar with sparse dataframe.  And the sparse item and NaN are quite similar and confuse.

```python
Traceback (most recent call last):
  File "/Users/facaiyan/Workshop/pandas/pandas/sparse/tests/test_sparse.py", line 966, in test_fill_value_missing_with_binary_operators
    tm.assert_sp_frame_equal(res, exp)
  File "/Users/facaiyan/Workshop/pandas/pandas/util/testing.py", line 1257, in assert_sp_frame_equal
    assert_sp_series_equal(series, right[col])
  File "/Users/facaiyan/Workshop/pandas/pandas/util/testing.py", line 1229, in assert_sp_series_equal
    assert_sp_array_equal(left.block.values, right.block.values)
  File "/Users/facaiyan/Workshop/pandas/pandas/util/testing.py", line 1205, in assert_sp_array_equal
    assert_almost_equal(left.sp_values, right.sp_values)
  File "/Users/facaiyan/Workshop/pandas/pandas/util/testing.py", line 124, in assert_almost_equal
    return _testing.assert_almost_equal(left, right, **kwargs)
  File "pandas/src/testing.pyx", line 58, in pandas._testing.assert_almost_equal (pandas/src/testing.c:3809)
  File "pandas/src/testing.pyx", line 120, in pandas._testing.assert_almost_equal (pandas/src/testing.c:2101)
  File "/Users/facaiyan/Workshop/pandas/pandas/util/testing.py", line 913, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

numpy array shapes are different
[left]:  (10,)
[right]: (9,)
```